resources/osl_mysql_test.rb

```diff
@@ -0,0 +1,38 @@
+provides :osl_mysql_test
+resource_name :osl_mysql_test
+unified_mode true
+
+default_action :create
+
+property :database, String, name_property: true
+property :username, String, required: true
+property :password, String, required: true
+property :server_password, String, default: lazy { 'osl_mysql_test' }
+property :encoding, String, default: lazy { 'utf8mb4' }
+property :collation, String, default: lazy { 'utf8mb4_unicode_ci' }
```

Comment:
`lazy` is not required here since you're directly setting the string, please remove that.